lightning/src/ln/channelmanager.rs

```diff
@@ -5038,13 +5155,25 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 			pending_payment.write(writer)?;
 		}
 
-		let pending_outbound_payments = self.pending_outbound_payments.lock().unwrap();
-		(pending_outbound_payments.len() as u64).write(writer)?;
-		for session_priv in pending_outbound_payments.iter() {
-			session_priv.write(writer)?;
+		let pending_outbounds_compat = {
+			let mut pending_session_privs: HashSet<[u8; 32]> = HashSet::new();
```

Comment:
Do we have to create a hashset here? I believe the actual session privs are guaranteed to be unique. We could instead just count them in the loop here instead and then loop twice, no?